base/broadcast.jl

```diff
@@ -423,6 +423,23 @@ end
     broadcastable(x)
 
 Return either `x` or an object like `x` such that it supports `axes` and indexing.
+
+If `x` supports iteration, the returned value should match [`collect(x)`](@ref).
```

Comment:
This isn't true.  e.g. `collect(3)` returns an `Array{Int,0}`, but `broadcastable` just returns `3`.

Reply:
Fair.  How about I add `with respect to axes and indexing`?